core/domain/rights_manager.py

```diff
@@ -538,316 +537,194 @@ def _get_activity_rights(activity_type, activity_id):
+def check_can_release_ownership(user, activity_rights):
+    """Checks whether the user can release ownership for given activity.
 
-    def can_publish(self, activity_type, activity_id):
-        """Returns whether this user is allowed to publish the given activity.
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        activity_rights: ActivityRights or None. Rights object for the given
+            activity.
 
-        Args:
-            activity_type: str. The type of activity. Possible values:
-                constants.ACTIVITY_TYPE_EXPLORATION
-                constants.ACTIVITY_TYPE_COLLECTION
-            activity_id: str. ID of the activity.
+    Returns:
+        bool. Whether the user can release ownership for given activity.
+    """
+    if activity_rights is None:
+        return False
 
-        Returns:
-            bool. Whether the user is allowed to publish the given activity.
-        """
-        activity_rights = _get_activity_rights(activity_type, activity_id)
-        if activity_rights is None:
-            return False
+    if activity_rights.is_private():
+        return False
 
-        if activity_rights.status != ACTIVITY_STATUS_PRIVATE:
-            return False
-        if activity_rights.cloned_from:
-            return False
+    return check_can_modify_activity_roles(
+        user, activity_rights)
 
-        return self.is_owner(activity_type, activity_id) or self.is_admin()
 
-    def can_unpublish(self, activity_type, activity_id):
-        """Returns whether this user is allowed to unpublish the given
-        activity.
+def check_can_publish_activity(user, activity_rights):
+    """Checks whether the user can publish given activity.
 
-        Args:
-            activity_type: str. The type of activity. Possible values:
-                constants.ACTIVITY_TYPE_EXPLORATION
-                constants.ACTIVITY_TYPE_COLLECTION
-            activity_id: str. ID of the activity.
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        activity_rights: ActivityRights or None. Rights object for the given
+            activity.
 
-        Returns:
-            bool. Whether the user is allowed to unpublish the given
-                activity.
-        """
-        activity_rights = _get_activity_rights(activity_type, activity_id)
-        if activity_rights is None:
-            return False
+    Returns:
+        bool. Whether the user can publish given activity.
+    """
+    if activity_rights is None:
+        return False
 
-        if activity_rights.status != ACTIVITY_STATUS_PUBLIC:
-            return False
-        if activity_rights.community_owned:
-            return False
+    if activity_rights.cloned_from:
+        return False
 
-        return self.is_moderator()
+    if role_services.ACTION_PUBLISH_ANY_ACTIVITY in user.actions:
```

Comment:
oh. I don't know, maybe I had a different picture of publishing back then. I even have a check for admin can get True from the decorator using this.

Updated and added test.